main.tf

```
@@ -26,6 +26,15 @@ resource "aws_acm_certificate" "this" {
     certificate_transparency_logging_preference = var.certificate_transparency_logging_preference ? "ENABLED" : "DISABLED"
   }
 
+  dynamic "validation_option" {
+    for_each = var.validation_option
+
+    content {
+      domain_name       = validation_option.key
```

maybe a slight tweak so that users who might not know its the key don't hit issues
```suggestion
      domain_name       = try(validation_option.value.domain_name, validation_option.key)
```